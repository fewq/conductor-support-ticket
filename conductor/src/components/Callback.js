import React, { Component } from "react";

class Callback extends Component {
  state = {};
  render() {
    return (
      <div>
        <h1>Callback component</h1>
        <h1>Loading...</h1>
      </div>
    );
  }
}

export default Callback;
