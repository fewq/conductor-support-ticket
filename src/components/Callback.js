import React, { Component } from "react";
import Auth from "../Auth";
import {renderLoading} from "./helper";

class Callback extends Component {
  state = {};

  componentDidMount() {
    const auth = new Auth();
    auth.handleAuthentication();
  }

  render() {
    return (
      <div>
        {renderLoading()}
      </div>
    );
  }
}

export default Callback;
