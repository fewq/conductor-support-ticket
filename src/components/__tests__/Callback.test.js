import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { render, cleanup } from 'react-testing-library'
import Callback from '../Callback';
import {toBeInTheDocument, toHaveClass} from 'jest-dom'
expect.extend({toBeInTheDocument, toHaveClass})

test('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Callback/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

test('renders greeting text', () => {
    const { getByText } = render(<Callback />)
    let loadingText = getByText(/Loading.../i);
    expect(loadingText).toBeInTheDocument();
  })

test('snapshot testing', () => {
    const tree = renderer
      .create(<Callback />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  