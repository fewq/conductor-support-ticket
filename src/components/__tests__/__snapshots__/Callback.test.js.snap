// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<div>
  <h1>
    Callback component
  </h1>
  <h1>
    Loading...
  </h1>
</div>
`;
